heron/metricscachemgr/tests/java/com/twitter/heron/metricscachemgr/metricscache/CacheCoreTestTreeIndex.java

```diff
@@ -0,0 +1,223 @@
+        new Comparator<MetricsCacheQueryUtils.MetricTimeRangeValue>() {
+          @Override
+          public int compare(MetricsCacheQueryUtils.MetricTimeRangeValue o1,
+                             MetricsCacheQueryUtils.MetricTimeRangeValue o2) {
+            return (int) (o1.startTime - o2.startTime);
+          }
+        };
+  }
+
+  /*
+   * query 1 bucket
+   */
+  @Test
+  public void test1() {
+    MetricsCacheQueryUtils.MetricRequest request = new MetricsCacheQueryUtils.MetricRequest();
+    request.startTime = now - 95 * 1000;
+    request.endTime = now - 75 * 1000;
+    request.aggregationGranularity = RAW;
+    MetricsCacheQueryUtils.MetricResponse response = cacheCore.getMetrics(request, metricsFilter);
+    // there is only one <component, instance, metric> tuple
+    Assert.assertEquals(response.metricList.size(), 1);
+    // there should be 2 metrics
+    List<MetricsCacheQueryUtils.MetricTimeRangeValue> list =
+        response.metricList.get(0).getMetricValue();
+    Assert.assertEquals(list.size(), 2);
+    // sort
+    list.sort(comparator);
+    // check values
+    Assert.assertEquals(list.get(0).value, "0.1");
+    Assert.assertEquals(list.get(1).value, "0.2");
+  }
+
+  /*
+   * query 2 buckets
+   */
+  @Test
+  public void test2() {
+    MetricsCacheQueryUtils.MetricRequest request = new MetricsCacheQueryUtils.MetricRequest();
+    request.startTime = now - 95 * 1000;
+    request.endTime = now - 45 * 1000;
+    request.aggregationGranularity = RAW;
+    MetricsCacheQueryUtils.MetricResponse response = cacheCore.getMetrics(request, metricsFilter);
+    // there is only one <component, instance, metric> tuple
+    Assert.assertEquals(response.metricList.size(), 1);
+    // there should be 4 metrics
+    List<MetricsCacheQueryUtils.MetricTimeRangeValue> list =
+        response.metricList.get(0).getMetricValue();
+    Assert.assertEquals(list.size(), 4);
+    // sort
+    list.sort(comparator);
+    // check value
+    Assert.assertEquals(list.get(0).value, "0.1");
+    Assert.assertEquals(list.get(1).value, "0.2");
+    Assert.assertEquals(list.get(2).value, "0.3");
+    Assert.assertEquals(list.get(3).value, "0.4");
+  }
+
+  /*
+   * query all buckets
+   */
+  @Test
+  public void testAll() {
+    MetricsCacheQueryUtils.MetricRequest request = new MetricsCacheQueryUtils.MetricRequest();
+    request.startTime = now - 200 * 1000;
+    request.endTime = now;
+    request.aggregationGranularity = RAW;
+    MetricsCacheQueryUtils.MetricResponse response = cacheCore.getMetrics(request, metricsFilter);
+    // there is only one <component, instance, metric> tuple
+    Assert.assertEquals(response.metricList.size(), 1);
+    // there should be 7 metrics
+    List<MetricsCacheQueryUtils.MetricTimeRangeValue> list =
+        response.metricList.get(0).getMetricValue();
+    Assert.assertEquals(list.size(), 7);
+    // sort
+    list.sort(comparator);
+    // check value
+    Assert.assertEquals(list.get(0).value, "0.1");
```

Comment:
please do these all in a loop, or better yet do them in a single helper method that get reused.